Module4/Assignment4/Assignment4.cpp

```diff
@@ -7,33 +7,187 @@
 #include <stdio.h>
 #include <math.h>
 #include <iostream>
+#include <string.h>
 using namespace std;
 
+bool isSorted(int *numbers, int size)
+{
+    if (size <= 1)
+    {
+        return true;
+    }
+    int previousElement = numbers[0];
+    for (int i = 1; i < size; i++)
+    {
+        if (previousElement > numbers[i])
+        {
+            return false;
+        }
+        previousElement = numbers[i];
+    }
+    return true;
+}
+void printArray(int *numbers, int size)
+{
+    for (int i = 0; i < size; i++)
+    {
+        cout << numbers[i] << ", ";
+    }
+    cout << endl;
+}
+
+const int MAX_SIZE = 128;
+int merge(int theArray[], int first, int mid, int last)
+{
+    int accessCount = 0;
+    int tempArray[MAX_SIZE];
+    memset(tempArray, 0xdeadbeef, sizeof(tempArray));
+    int first1 = first;   // begin first subarray
+    int last1 = mid;      // end first subarray
+    int first2 = mid + 1; // begin second subarray
+    int last2 = last;     // end second subarray
+    int index = first1;
+    while ((first1 <= last1) && (first2 <= last2))
+    {
+        accessCount += 2;
+        if (theArray[first1] <= theArray[first2])
+        {
+            tempArray[index] = theArray[first1];
+            accessCount++;
+            first1++;
+        }
+        else
+        {
+            tempArray[index] = theArray[first2];
+            accessCount++;
+            first2++;
+        } // end if
```

Comment:
I approve of these sorts of comments. For very long ones it's good to say *which* if or while they're concluding: `// end if for comparing array elements` or whatever